public-roadmap/index.ts

```diff
@@ -39,18 +42,58 @@ export const endpoint: resolvers.Root["endpoint"] = async (req) => {
       const { id, name, description: heightDesc, status: heightStatus } = task;
       const status = formatStatus(heightStatus);
       const description = formatDescription(heightDesc);
-
       return { id, name, description, status };
     });
 
-  switch (`${req.method} ${req.path}`) {
-    case "GET /":
-      const body = JSON.stringify(publicRoadmapTasks);
+  return publicRoadmapTasks;
+}
+
+function fetchUpvotes({
+  ipAddress,
+  taskId,
+}: {
+  ipAddress: string;
+  taskId?: string;
+}) {
+  if (!taskId) return { numUpvotes: 0, hasUpvoted: false };
+
+  if (!state.tasks[taskId]) {
+    state.tasks[taskId] = {};
+  }
+
+  const numUpvotes = Object.keys(state.tasks[taskId]).length;
+  const hasUpvoted = state.tasks[taskId][ipAddress] || false;
+
+  return { numUpvotes, hasUpvoted };
+}
+
+export const endpoint: resolvers.Root["endpoint"] = async (req) => {
+  const ipAddress = JSON.parse(req.headers)["x-forwarded-for"];
+
+  switch (req.method) {
```

Comment:
```suggestion
  switch (`${req.method} ${req.path}`) {
```

This is a pattern I've used a few times that makes a simple, ergonomic router. So your case statements would be something like `case "GET /upvotes":`.

Reply:
I initially was using that pattern, but most paths include a task id, so the `case` expression strict equality doesn't work. I did some refactoring just now to clean up the routing. Lmk what you think.